### PR TITLE
[release 1.22] switch to using KUBERNETES_VERSION from version.sh for windows

### PR DESCRIPTION
**Pull request by luthermonson**
#### Proposed Changes ####
switch to using the build arg KUBERNETES_VERSION over the hardcoded ENV in the windows dockerfile

#### Linked Issues ####
https://github.com/rancher/rke2/issues/1744

### CODE REVIEW

File: scripts/build-image-runtime [hunk 1]
```diff
@@ -1,34 +1 @@
...
    --build-arg CACHEBUST="$(date +%s%N)" \
    --tag ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION}-windows-amd64 \
    --target windows-runtime \
    --file Dockerfile.windows \
    .

mkdir -p build/images
docker image save \
    --output build/images/${PROG}-runtime.tar \
    ${REPO}/${PROG}-runtime:${DOCKERIZED_VERSION}-${GOOS}-${GOARCH}
```
Comment (dweomer) [inline scripts/build-image-runtime:1]: @luthermonson can you `dos2unix` this please

Comment (briandowns) [inline scripts/build-image-runtime:1]: Can you update the format here?

File: Dockerfile.windows [hunk 2]
```diff
@@ -67,20 +68,20 @@ RUN cat sha256sum.txt | head -n1 | awk '{print $1"  wins.exe"}' >> wins.exe.sha2
 RUN sha256sum -c wins.exe.sha256
 RUN mv wins.exe rancher/
 
-RUN curl -sLO https://dl.k8s.io/release/${KUBERNETES_VERSION}/bin/windows/amd64/kubectl.exe
-RUN curl -sLO https://dl.k8s.io/${KUBERNETES_VERSION}/bin/windows/amd64/kubectl.exe.sha256
+RUN curl -sLO https://dl.k8s.io/release/$KUBERNETES_VERSION/bin/windows/amd64/kubectl.exe
```
Comment (briandowns) [inline Dockerfile.windows:71]: What's the benefit of removing the curly braces?

File: scripts/build-image-runtime (hunk 1 above)
Comment (luthermonson) [reply scripts/build-image-runtime:1]: this file wasnt suppose to be added it was just me testing in wsl, this is still a wip guys...

File: Dockerfile.windows (hunk 2 above)
Comment (luthermonson) [reply Dockerfile.windows:71]: trying to confirm env vs arg, dockerfile syntax vs shell syntax... we had issues in drone files i just dont remember which way it works yet. still a wip and i can't get it to build locally

Comment (dweomer) [reply Dockerfile.windows:71]: > trying to confirm env vs arg, dockerfile syntax vs shell syntax... we had issues in drone files i just dont remember which way it works yet. still a wip and i can't get it to build locally

should be safe to leave the braces alone

**Review by dweomer — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Pull request merged**